### DISCUSSION TITLE
How to get a TOTAL for a column of a table?

### DISCUSSION THREAD

**Discussion by HeronCS** [Budibase/budibase#6278] [Help] [answered]
Hi,

How would I create a total amount displayed, for each column of a table?

The values are both Static and Dynamic (through formula column in data).

**Comment by melohagan** [comment]
Hey @HeronCS 

You can use the Sum calculation via views: https://docs.budibase.com/docs/views

**Reply by HeronCS** [reply]
This is NOT what I'm after, this is a very time consuming task.

**Comment by HeronCS** [comment]
I shall illustrate what I mean.
![total](https://user-images.githubusercontent.com/86607231/173395959-8ecf2269-c751-40f5-ada3-95f0a4b2e01b.PNG)

**Reply by HeronCS** [reply]
I have twelve months, and MANY years. So using your first suggested. Would be impracticable to achieving this goal. I already have a View for each year. And then filtered on the Design, by Date (Month column). I then copied twelve times. I can then just copy 2022 to 2023 and so on, and link each year.

**Reply by HeronCS** [reply]
To achieve the complete goal, is to filter by Subcontractors, which I cannot do. As it's a Relationship. So Each Subcontractor has his/her years > months that can be printed off for manual verification.

**Reply by melohagan** [reply]
Thank you @HeronCS I see what you mean, apologies - I did send you how to sum rows, not columns.

I see you did ask this before: https://github.com/Budibase/budibase/discussions/5729
The answer did link the current solution which is a workaround: https://github.com/Budibase/budibase/discussions/3071#discussioncomment-1498568

There's also a check list item in this issue for adding both column and row aggregations: https://github.com/Budibase/budibase/issues/3273

**Reply by HeronCS** [reply]
I've answered #5729 to here, as the illustration helps better understand the intent.

**Reply by HeronCS** [reply]
Until #3273 is completed, I might as well go back to spreadsheets and iframe them in. This is top priority for any/all of our projects with budibase.

**Comment by j-wil** [comment, marked as answer]
You can use JS functions to accomplish this. 

```js
const arr = $("Total Payments.Rows")

let total = 0;

arr.forEach(obj => {
  if (obj['amount'] !== undefined) {
    total += obj['amount'];
  }
});

return total
```

Just replace `$("Total Payments.Rows")` with the table and `obj['amount']` with the column

**Reply by melohagan** [reply]
You can also use `reduce` in JS: https://stackoverflow.com/questions/1230233/how-to-find-the-sum-of-an-array-of-numbers

Thanks for sharing!
